### PR TITLE
Disable source build cache on pocl/nvidia

### PR DESCRIPTION
**Pull request by matthiasdiener**
Followup of #738.

_Please squash_

### CODE REVIEW

**Comment by matthiasdiener** [comment]
~~[1a9ad0d](https://github.com/inducer/pyopencl/pull/749/commits/1a9ad0dc89f1d66df750f71374fcb14604eec0cd) outlines a way to ensure pocl uses PID-local caches.~~ This is ready for a first look @inducer @majosm

File: pyopencl/__init__.py [hunk 1]
```diff
@@ -1540,6 +1548,10 @@ def choose_devices(interactive: Optional[bool] = None,
     :returns: a list of :class:`Device` instances.
     """
 
+    # This must be done before PoCL is actually loaded
+    from pyopencl.characterize import setup_pocl_cache_dirs
+    setup_pocl_cache_dirs()
```
Comment (inducer) [inline pyopencl/__init__.py]: Could you explain why this is here? This seems to defeat inter-run caching, which, in my book, is definitely desirable.

Comment (matthiasdiener) [reply pyopencl/__init__.py]: This is a workaround for the intermittent failures we have seen as part of #738. 

I couldn't think of a way to ensure PID-local caches that also support inter-run caching, without resorting to using MPI rank numbers (which wouldn't even help for the CI errors seen in #738).

Edit: some sources regarding caching issues in pocl: https://github.com/pocl/pocl/issues/430, https://github.com/pocl/pocl/pull/466

Comment (inducer) [reply pyopencl/__init__.py]: Before we go in this direction, I think we should engage with the pocl folks and least file an issue to see what they think about fixing concurrent access to their caches.

Comment (matthiasdiener) [reply pyopencl/__init__.py]: Reported here: https://github.com/pocl/pocl/issues/1480, including a small reproducer.

Comment (matthiasdiener) [reply pyopencl/__init__.py]: Reverted in [b368eba](https://github.com/inducer/pyopencl/pull/749/commits/b368eba29f942b3e1939489f131a3799bbfbe37e)

**Comment by matthiasdiener** [comment]
This is ready for review @inducer. I have not noticed any cache concurrency issues with this PR+the new pocl version (`5.0 * _5`).

**Pull request merged**